libkbfs/ops.go

```diff
@@ -867,10 +867,7 @@ func addToCollapsedWriteRange(writes []WriteRange,
 			// Truncate past the last write.
 			return append(head, wNew)
 		} else if mid[0].isTruncate() {
-			// Min truncate wins
-			if mid[0].Off < wNew.Off {
-				return append(head, mid[0])
-			}
+			// Later truncates always win.
```

Comment:
Good question, it is filled with `\0` in that case.  These write ranges aren't used for the actual data, that still lives in the block.  The write ranges are just describing the data so that things like CR can figure out if a file with two different sets of operations on them need conflict resolution, or if they are still essentially the same file.

You're right though that a more accurate description in that case would be to add a `WriteRange{Len=1000, off=1000}` to reflect the fact that those bytes changed.  I'll update this and add a test when I get a chance.